pkg/activator/util/retryer.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2018 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import "time"
+
+type Retryer interface {
+	Retry(func() bool) int
+}
+
+type ActionFunc func() bool
+type RetryerFunc func(ActionFunc) int
+
+func (r RetryerFunc) Retry(f func() bool) int {
+	return r(f)
+}
+
+// NewLinearRetryer will return a retryer that retries `action` up to
+// `maxRetries` times with `interval` delay between retries
+func NewLinearRetryer(interval time.Duration, maxRetries int) Retryer {
```

Comment:
@markusthoemmes is adding exponential backoff in https://github.com/knative/serving/pull/1814 and I've suggested to @pivotal-sukhil-suresh to reuse that code in https://github.com/knative/serving/pull/1665#pullrequestreview-145313089.  This would be a good way to do that reuse, with an ExponentialRetrier.

So there are three changes in-flight in this space. I'm not sure what order they should land in.  Any suggestions?

Reply:
I'd suggest to land this one first. Then I can base my PR on top of it (I felt bad because I had no tests anyway) and then @pivotal-sukhil-suresh can base off of mine to gain the ability of having an exponential backoff. WDYT @josephburnett

Reply:
That would be fine.